src/pybind/mgr/dashboard/tests/__init__.py

```diff
@@ -140,32 +140,43 @@ def __init__(self, *args, **kwargs):
         PLUGIN_MANAGER.hook.configure_cherrypy(config=cherrypy.config)
         super(ControllerTestCase, self).__init__(*args, **kwargs)
 
-    def _request(self, url, method, data=None, headers=None):
+    def _request(self, url, method, data=None, headers=None, version=None):
         if not data:
             b = None
-            h = None
+            if version:
+                h = [('Accept', 'application/vnd.ceph.api.v{}+json'.format(version)),
+                     ('Content-Type', 'application/vnd.ceph.api.v{}+json'.format(version)),
+                     ('Content-Length', '0')]
+            else:
+                h = None
         else:
             b = json.dumps(data)
-            h = [('Content-Type', 'application/json'),
-                 ('Content-Length', str(len(b)))]
+            if version:
+                h = [('Accept', 'application/vnd.ceph.api.v{}+json'.format(version)),
+                     ('Content-Type', 'application/vnd.ceph.api.v{}+json'.format(version)),
+                     ('Content-Length', str(len(b)))]
+            
+            else:
+                h = [('Content-Type', 'application/json'),
+                     ('Content-Length', str(len(b)))]
         if headers:
             h = headers
         self.getPage(url, method=method, body=b, headers=h)
```

Comment:
The body received is empty even if some data is passed to the request, not sure what's the problem.